### DISCUSSION TITLE
include device ID in POST message

### DISCUSSION THREAD

**Discussion by requireiot** [toverainc/willow#120] [Ideas]
I want to use multiple Willow boxes, so my home automation engine needs to know which device sent a message. (I am using 2x ESP32-S3-Box-mini, with Multinet local speech recognition enabled, and the generic REST endpoint).

My current solution: in `main/main.c`, ca. line 179, change from
```
  json = calloc(sizeof(char), 29 + strlen(lookup_cmd_multinet(command_id)));
  snprintf(json, 29 + strlen(lookup_cmd_multinet(command_id)), "{\"text\":\"%s\",\"language\":\"en\"}",
      lookup_cmd_multinet(command_id));
```
to 
```
   size_t jsonlen = 29 + strlen(lookup_cmd_multinet(command_id)) + 40;
   json = calloc(sizeof(char), jsonlen);
   snprintf(json, jsonlen, "{\"text\":\"%s\",\"language\":\"en\",\"siteId\":\"%s\"}",
      lookup_cmd_multinet(command_id), CONFIG_LWIP_LOCAL_HOSTNAME);
```
and then, before I flash each box, I change the CONFIG_LWIP_LOCAL_HOSTNAME parameter via `utils.sh config`

**Comment by kristiankielhofner** [comment]
We have begun implementing a broad approach to this that comes in two forms:

1) A Willow "group ID" for associated devices used in the same environment. This will be used for functionality such as #74.

2) A Willow "device ID". We will shortly have the ability to essentially use CONFIG_LWIP_LOCAL_HOSTNAME as a hostname base with the last four octets of the MAC address appended on boot. Like the group ID we will also support attaching this as a header and/or potentially in the JSON from the device as you have done.
